public/app/core/services/echo/echoConsumerFactory.ts

```diff
@@ -0,0 +1,5 @@
+import { EchoEvent, EchoConsumerCtor, EchoConsumer } from './types';
+import { getEcho } from './EchoSrv';
+
+export const echoConsumerFactory = <O, T extends EchoEvent>(ctor: EchoConsumerCtor<T, O>, opts: O): EchoConsumer<T> =>
+  new ctor(getEcho(), opts);
```

Comment:
Is this a necessary complexity?

Reply:
Yeah, probably can be removed. Added it primarily to hide the fact of passing Echo instance to the constructor